internal/provider/resource_gitlab_pages_domain.go

```diff
@@ -0,0 +1,220 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_pages_domain", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_pages_domain`" + ` resource enables endpoints for connecting custom domain(s) and TLS certificates in GitLab Pages.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/pages_domains.html)`,
+
+		CreateContext: resourceGitlabPagesDomainCreate,
+		ReadContext:   resourceGitlabPagesDomainRead,
+		UpdateContext: resourceGitlabPagesDomainUpdate,
+		DeleteContext: resourceGitlabPagesDomainDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Description: "The custom domain.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"project": {
+				Description: "The ID or full path of the project which the branch is created against.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"auto_ssl_enabled": {
+				Description: "Enables automatic generation of SSL certificates issued by Let’s Encrypt for custom domains.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
+			},
+			"certificate": {
+				Description: "The certificate in PEM format with intermediates following in most specific to least specific order.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
```

Comment:
NIT: we usually do not specify the defaults

```suggestion
```

Reply:
I'd leave the ForceNew here too. I might be wrong, but the `certificate_data` is a computed property based on the `certificate`. The plan might be misleading without ForceNew, as `certificate_data` will be known only after the apply step. I guess ForceNew might achieve this.

Reply:
I don't get that 😅  ... I mean `ForceNew: false` is the default anyways, I'm just saying we should remove from the schema - so that it's non-explicit.